### PR TITLE
feat: enable unsigned transactions in tracing

### PR DESCRIPTION
**Pull request by skosito**
Some of transactions are applied directly (fungible module). They don't have a signature, and tracing those txs fails with signature error. This fix is checking for unsigned txs and is not using it to parse `from` field.

### CODE REVIEW

File: x/evm/keeper/state_transition.go [hunk 1]
```diff
@@ -157,6 +158,8 @@ func (k *Keeper) ApplyTransaction(ctx sdk.Context, msgEth *types.MsgEthereumTx)
 		return nil, errorsmod.Wrap(err, "failed to load evm config")
 	}
 	ethTx := msgEth.AsTransaction()
+	fmt.Println("regular ethermint tx", ethTx.Hash())
```
Comment (lumtis) [inline x/evm/keeper/state_transition.go:161]: Is it kept for debugging purposes? Why not use `ctx.Logger`?

File: x/evm/keeper/grpc_query.go [hunk 2]
```diff
@@ -424,12 +424,35 @@ func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*typ
...
-		msg, err := ethTx.AsMessage(signer, cfg.BaseFee)
+		var msg ethtypes.Message
+		// if tx is not unsigned, from field should be derived from signer, which can be done using AsMessage function
+		if !isUnsigned(ethTx) {
+			signer := ethtypes.MakeSigner(cfg.ChainConfig, big.NewInt(ctx.BlockHeight()))
+			msg, err = ethTx.AsMessage(signer, cfg.BaseFee)
+			if err != nil {
+				continue
+			}
+			// tx.From = m.From().String()
```
Comment (lumtis) [inline x/evm/keeper/grpc_query.go:440]:
```suggestion
```

File: x/evm/keeper/grpc_query.go [hunk 3]
```diff
@@ -424,12 +424,35 @@ func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*typ
...
+				ethTx.To(),
+				ethTx.Nonce(),
+				ethTx.Value(),
+				ethTx.Gas(),
+				new(big.Int).Set(ethTx.GasPrice()),
+				new(big.Int).Set(ethTx.GasFeeCap()),
+				new(big.Int).Set(ethTx.GasTipCap()),
+				ethTx.Data(),
+				ethTx.AccessList(),
+				false, // isFake
```
Comment (lumtis) [inline x/evm/keeper/grpc_query.go:453]: Why is this used here? Can we provide more context?

File: x/evm/keeper/grpc_query.go [hunk 4]
```diff
@@ -470,6 +496,12 @@ func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*typ
 	}, nil
 }
 
+func isUnsigned(ethTx *ethtypes.Transaction) bool {
+	r, v, s := ethTx.RawSignatureValues()
+
+	return (r == nil && v == nil && s == nil) || (r.Cmp(big.NewInt(0)) == 0 && v.Cmp(big.NewInt(0)) == 0 && s.Cmp(big.NewInt(0)) == 0)
```
Comment (lumtis) [inline x/evm/keeper/grpc_query.go:502]:
```suggestion
r.Cmp(big.NewInt(0)) == 0 
```
I think `r.Int64() == 0` is simpler

File: x/evm/keeper/grpc_query.go [hunk 5]
```diff
@@ -454,7 +477,10 @@ func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*typ
 		_ = json.Unmarshal([]byte(req.TraceConfig.TracerJsonConfig), &tracerConfig)
```
Comment (lumtis) [inline x/evm/keeper/grpc_query.go:477]: I know it's forked code but wondering if we shouldn't return an error here

File: x/evm/keeper/grpc_query.go [hunk 6]
```diff
@@ -550,7 +584,29 @@ func (k *Keeper) traceTx(
...
+		tx.Nonce(),
+		tx.Value(),
+		tx.Gas(),
+		new(big.Int).Set(tx.GasPrice()),
+		new(big.Int).Set(tx.GasFeeCap()),
+		new(big.Int).Set(tx.GasTipCap()),
+		tx.Data(),
+		tx.AccessList(),
+		false, // isFake
+	)
```
Comment (lumtis) [inline x/evm/keeper/grpc_query.go:608]: Seems like we have twice this logic. Maybe we can create a function for it?

File: x/evm/types/events.go [hunk 7]
```diff
@@ -29,6 +29,10 @@ const (
 	AttributeKeyTxGasUsed       = "txGasUsed"
 	AttributeKeyTxType          = "txType"
 	AttributeKeyTxLog           = "txLog"
+
+	AttributeKeyTxNonce = "txNonce"
+	AttributeKeyTxData  = "txData"
```
Comment (kingpinXD) [inline x/evm/types/events.go:34]: Where is this used ?

**Review by kingpinXD — APPROVED** [review]
lgtm , minor comment for my understanding

**Review by lumtis — APPROVED** [review]
LGTM, if we can use semantic PR for the title it would be great

**Pull request merged**